executor/runtime/docker/capabilities_test.go

```diff
@@ -58,14 +65,80 @@ func TestOverrideFuseProfile(t *testing.T) {
 	assert.NoError(t, err)
 	hostConfig := container.HostConfig{}
 
-	assert.True(t, c.FuseEnabled())
 	assert.NotNil(t, c.AppArmorProfile())
 	assert.Equal(t, *c.AppArmorProfile(), "docker_foo")
 
-	assert.NoError(t, setupAdditionalCapabilities(c, &hostConfig))
+	assert.NoError(t, setupAdditionalCapabilities(c, &hostConfig, runtimeTypes.MainContainerName))
 
 	assert.Contains(t, hostConfig.CapAdd, "SYS_ADMIN")
 	assert.Len(t, hostConfig.CapDrop, 0)
 	assert.Len(t, hostConfig.SecurityOpt, 2)
 	assert.Contains(t, hostConfig.SecurityOpt, "apparmor:docker_foo")
+
+	fuseMapping := container.DeviceMapping{
+		PathOnHost:        fuseDev,
+		PathInContainer:   fuseDev,
+		CgroupPermissions: "rmw",
+	}
+	assert.Contains(t, hostConfig.Resources.Devices, fuseMapping)
+}
+
+func TestImageBuildingProfile(t *testing.T) {
+	pod, conf, err := runtimeTypes.PodContainerTestArgs()
+	assert.NoError(t, err)
+
+	ann := podCommon.ContainerAnnotation(runtimeTypes.MainContainerName, podCommon.AnnotationKeySuffixContainersCapabilities)
+	pod.Annotations[ann] = titusAPI.ContainerCapability_ContainerCapabilityImageBuilding.String()[len("ContainerCapability"):]
+
+	c, err := runtimeTypes.NewPodContainer(pod, &sync.Mutex{}, *conf)
+	assert.NoError(t, err)
+	hostConfig := container.HostConfig{}
+
+	assert.NoError(t, setupAdditionalCapabilities(c, &hostConfig, runtimeTypes.MainContainerName))
+
+	assert.Contains(t, hostConfig.CapAdd, "SYS_ADMIN")
+	assert.Len(t, hostConfig.CapDrop, 0)
+	assert.Len(t, hostConfig.SecurityOpt, 2)
+	assert.Contains(t, hostConfig.SecurityOpt, "apparmor:docker_image_building")
+
+	// shouldn't have the fuse device
+	fuseMapping := container.DeviceMapping{
+		PathOnHost:        fuseDev,
+		PathInContainer:   fuseDev,
+		CgroupPermissions: "rmw",
+	}
+	assert.NotContains(t, hostConfig.Resources.Devices, fuseMapping)
+}
+
+func TestImageBuildingAndFUSEProfile(t *testing.T) {
+	pod, conf, err := runtimeTypes.PodContainerTestArgs()
+	assert.NoError(t, err)
+
+	ann := podCommon.ContainerAnnotation(runtimeTypes.MainContainerName, podCommon.AnnotationKeySuffixContainersCapabilities)
+	caps := fmt.Sprintf("%s,%s",
+		titusAPI.ContainerCapability_ContainerCapabilityImageBuilding.String()[len("ContainerCapability"):],
+		titusAPI.ContainerCapability_ContainerCapabilityFUSE.String()[len("ContainerCapability"):])
+	pod.Annotations[ann] = caps
+
+	c, err := runtimeTypes.NewPodContainer(pod, &sync.Mutex{}, *conf)
+	assert.NoError(t, err)
+	hostConfig := container.HostConfig{}
+
+	assert.NoError(t, setupAdditionalCapabilities(c, &hostConfig, runtimeTypes.MainContainerName))
+
+	assert.Contains(t, hostConfig.CapAdd, "SYS_ADMIN")
+	assert.Len(t, hostConfig.CapDrop, 0)
+	assert.Len(t, hostConfig.SecurityOpt, 2)
+
+	// image building is a superset of fuse, so we use that profile when
+	// both are supplied
+	assert.Contains(t, hostConfig.SecurityOpt, "apparmor:docker_image_building")
```

Comment:
You are going above and beyond here.
For now lets kiss and only handle one capability at a time here.